jaseci_serv/jaseci_serv/base/tests/test_orm_hooks.py

```diff
@@ -315,3 +315,186 @@ def test_fast_edges_with_context(self):
         cedge = snode.outbound_edges()[0]
         self.assertIn("a", cedge.context.keys())
         self.assertEqual(cedge.context["a"], "b")
+
+    def test_node_saves_on_traversing_graphs(self):
+        user = self.user
+        hook = user._h = JsOrc.hook()
+        gph = Graph(m_id=0, h=hook)
+        sent = Sentinel(m_id=0, h=gph._h)
+        sent.register_code(jtc.simple_graph)
+        test_walker = sent.run_architype("sample")
+        test_walker.prime(gph)
+        test_walker.run()
+
+        self.assertSetEqual(
+            set(
+                [
+                    "basic::sentinel",
+                    "root::graph",
+                    "a1::node",
+                    "a2::node",
+                    "a3::node",
+                    "b1::node",
+                    "b2::node",
+                    "root::architype",
+                    "a1::architype",
+                    "a2::architype",
+                    "a3::architype",
+                    "b1::architype",
+                    "b2::architype",
+                    "e1::architype",
+                    "e2::architype",
+                    "e3::architype",
+                    "generic::architype",
+                    "sample::architype",
+                    "sample2::architype",
+                    "sample3::architype",
+                    "generic::architype",
+                ]
+            ),
+            set([item.name + "::" + item.j_type for item in hook.save_obj_list]),
+        )
+
+        hook.commit()
+        hook.clear_cache()
+
+        test_walker = sent.run_architype("sample")
+        test_walker.prime(gph)
+        test_walker.run()
+
+        self.assertFalse(hook.save_obj_list)
+
+        test_walker = sent.run_architype("sample2")
+        test_walker.prime(gph)
+        test_walker.run()
+
+        self.assertSetEqual(
+            set(["b2::node", "a3::node"]),
+            set([item.name + "::" + item.j_type for item in hook.save_obj_list]),
+        )
+
+        hook.commit()
+        hook.clear_cache()
+
+        test_walker = sent.run_architype("sample3")
+        test_walker.prime(gph)
+        test_walker.run()
+
+        self.assertSetEqual(
+            set(["a1::node", "a2::node", "a3::node"]),
+            set([item.name + "::" + item.j_type for item in hook.save_obj_list]),
+        )
+
+        hook.commit()
+        hook.clear_cache()
+
+        test_walker = sent.run_architype("sample3")
+        test_walker.prime(gph)
+        test_walker.run()
+
+        # no a3::node since it was already disconnected
```

Comment:
Even it is disconnted, `a3` should still be saved right? Because you are modifying a3 directly with `a3: here.val = 2;`

Reply:
the second trigger of `sample3` will not include a3 as it was already disconnected from first `sample3` trigger